### PR TITLE
Add maintenance window

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: cloudwatch_events/templates/jenkins_maintenance_event_window_pattern.json.tpl [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  ],
+  "detail": {
+    "status": [
+      "FAILED"
+    ],
+    "window-id": [
+      "${window_id}"
+    ]
+  }
+}
```
Comment (TomJKing) [inline cloudwatch_events/templates/jenkins_maintenance_event_window_pattern.json.tpl:16]: Missing line break

File: cloudwatch_events/variables.tf [hunk 2]
```diff
@@ -8,3 +8,12 @@ variable "lambda_event_target_arn" {
...
 }
+variable "rule_name" {}
+variable "rule_description" {
+  default = ""
+}
+variable "event_variables" {
+  type        = map(string)
+  default     = {}
+  description = "A map of variables to pass to specific event patterns"
+}
```
Comment (TomJKing) [inline cloudwatch_events/variables.tf:19]: Missing line break

File: ssm_maintenance_window/data.tf [hunk 3]
```diff
@@ -0,0 +1,6 @@
+data aws_instance "target_instance" {
+  filter {
+    name   = "tag:Name"
+    values = [var.instance_name]
+  }
+}
```
Comment (TomJKing) [inline ssm_maintenance_window/data.tf:6]: Missing line break

File: ssm_maintenance_window/main.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
+data aws_caller_identity "current" {}
+
```
Comment (TomJKing) [inline ssm_maintenance_window/main.tf:2]: Extra line break

File: ssm_maintenance_window/main.tf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  task_invocation_parameters {
+    run_command_parameters {
+      parameter {
+        name   = "commands"
+        values = [var.command]
+      }
+    }
+  }
+}
```
Comment (TomJKing) [inline ssm_maintenance_window/main.tf:42]: Missing line break

File: ssm_maintenance_window/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,3 @@
+output "window_id" {
+  value = aws_ssm_maintenance_window.maintenance_window.id
+}
```
Comment (TomJKing) [inline ssm_maintenance_window/outputs.tf:3]: Missing line break

File: ssm_maintenance_window/variables.tf [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+}
+variable "cutoff" {
+  default = 0
+}
+variable "instance_name" {
+  description = "The name of the target instance to run the command against"
+}
+variable "command" {
+  description = "The command to execute in the task executed by the maintenance window"
+}
```
Comment (TomJKing) [inline ssm_maintenance_window/variables.tf:14]: Missing line break

**Review by TomJKing — COMMENTED** [review]
Very minor comments

File: ssm_maintenance_window/variables.tf (hunk 7 above)
Comment (MancunianSam) [reply ssm_maintenance_window/variables.tf]: Sorry about all the line breaks. I try to check better than this usually. I've fixed them all.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**